module-system/module-implementations/sov-accounts/src/call.rs

```diff
@@ -72,3 +72,23 @@ impl<C: sov_modules_api::Context> Accounts<C> {
         Ok(())
     }
 }
+
+#[cfg(all(feature = "arbitrary", feature = "native"))]
+impl<'a, C> arbitrary::Arbitrary<'a> for CallMessage<C>
+where
+    C: Context,
+    C::PrivateKey: arbitrary::Arbitrary<'a>,
+{
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        use sov_modules_api::PrivateKey;
+
+        let secret = C::PrivateKey::arbitrary(u)?;
+        let public = secret.pub_key();
+
+        let payload_len = u.arbitrary_len::<u8>()?;
+        let payload = u.bytes(payload_len)?;
+        let signature = secret.sign(payload);
+
+        Ok(Self::UpdatePublicKey(public, signature))
```

Comment:
How about generating some invalid signatures as well? We can choose based on an arbitrary `bool`.

Reply:
I think we should create a different structure to wrap randomly correct signatures. Otherwise, we can't consume the generated structs from here with confidence, and that would not allow us to expect certain behavior from arbitrary signatures. Wdyt?

Reply:
Tracked here to unblock this PR: https://github.com/Sovereign-Labs/sovereign-sdk/issues/747 🎉